crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -825,47 +834,60 @@ impl<'w> EntityWorldMut<'w> {
         entities.set(entity.index(), new_location);
     }
 
-    /// Remove the components of `bundle_info` from `entity`, where `self_location` and `old_location`
-    /// are the location of this entity, and `self_location` is updated to the new location.
+    /// Remove the components of `bundle` from `entity`.
     ///
-    /// SAFETY: `old_location` must be valid and the components in `bundle_info` must exist.
+    /// SAFETY: The components in `bundle_info` must exist.
     #[allow(clippy::too_many_arguments)]
-    unsafe fn remove_bundle_info(
-        entity: Entity,
-        self_location: &mut EntityLocation,
-        old_location: EntityLocation,
-        bundle_info: &BundleInfo,
-        archetypes: &mut Archetypes,
-        storages: &mut Storages,
-        components: &Components,
-        entities: &mut Entities,
-        removed_components: &mut RemovedComponentEvents,
-    ) {
+    unsafe fn remove_bundle(&mut self, bundle: BundleId) -> EntityLocation {
+        let entity = self.entity;
+        let world = &mut self.world;
+        let location = self.location;
+        let bundle_info = world.bundles.get_unchecked(bundle);
+
         // SAFETY: `archetype_id` exists because it is referenced in `old_location` which is valid
         // and components in `bundle_info` must exist due to this functions safety invariants.
```

Comment:
This safety comment needs to be updated, I suppose it means `location` now?

```suggestion
        // SAFETY: `archetype_id` exists because it is referenced in `location` which is valid
        // and components in `bundle_info` must exist due to this functions safety invariants.
```